flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerDriverTest.java

```diff
@@ -1,4 +1,598 @@
+				addContainerRequestFutures.get(addContainerRequestFuturesNumCompleted.getAndIncrement()).complete(null);
+				((YarnResourceManagerDriver) getDriver()).getYarnContainerEventHandler()
+					.onContainersAllocated(Collections.singletonList(testingContainer));
+			});
+			testingYarnAMRMClientAsyncBuilder.setRemoveContainerRequestConsumer((ignored1, ignored2) -> removeContainerRequestFuture.complete(null));
+			testingYarnAMRMClientAsyncBuilder.setReleaseAssignedContainerConsumer((ignored1, ignored2) -> releaseAssignedContainerFuture.complete(null));
+			testingYarnNMClientAsyncBuilder.setStartContainerAsyncConsumer((ignored1, ignored2, ignored3) -> startContainerAsyncFuture.complete(null));
+			testingYarnNMClientAsyncBuilder.setStopContainerAsyncConsumer((ignored1, ignored2, ignored3) -> stopContainerAsyncFuture.complete(null));
+			resourceEventHandlerBuilder.setOnWorkerTerminatedConsumer((ignore1, ignore2) -> getDriver().requestResource(testingTaskExecutorProcessSpec));
+		}
+
+		@Override
+		protected ResourceManagerDriver<YarnWorkerNode> createResourceManagerDriver() {
+			return new TestingYarnResourceManagerDriver(
+				flinkConfig,
+				env,
+				null,
+				"localhost:9000",
+				testingYarnAMRMClientAsyncBuilder,
+				testingYarnNMClientAsyncBuilder);
+		}
+
+		@Override
+		protected void validateInitialization() {
+			assertNotNull(((YarnResourceManagerDriver) getDriver()).getResourceManagerClient());
+			assertNotNull(((YarnResourceManagerDriver) getDriver()).getNodeManagerClient());
+		}
+
+		@Override
+		protected void validateWorkersRecoveredFromPreviousAttempt(Collection<YarnWorkerNode> workers) {
+			Assume.assumeTrue(HadoopUtils.isMinHadoopVersion(2, 2));
+			assertThat(workers.size(), is(1));
+
+			final ResourceID resourceId = workers.iterator().next().getResourceID();
+			assertThat(resourceId.toString(), is(testingContainer.getId().toString()));
+		}
+
+		@Override
+		protected void validateTermination() {
+			assertThat(((YarnResourceManagerDriver) getDriver()).getNodeManagerClient().getServiceState(), is(Service.STATE.STOPPED));
+			assertThat(((YarnResourceManagerDriver) getDriver()).getResourceManagerClient().getServiceState(), is(Service.STATE.STOPPED));
+		}
+
+		@Override
+		protected void validateDeregisterApplication() throws Exception {
+			assertThat(stopAndCleanupClusterFuture.get(TIMEOUT_SEC, TimeUnit.SECONDS), is("finish"));
+		}
+
+		@Override
+		protected void validateRequestedResources(Collection<TaskExecutorProcessSpec> taskExecutorProcessSpecs) throws Exception {
+			assertThat(taskExecutorProcessSpecs.size(), is(1));
+
+			final Resource resource = createTaskManagerContainerFuture.get(TIMEOUT_SEC, TimeUnit.SECONDS);
+
+			assertThat(resource.getMemory(), is(YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB));
+			assertThat(resource.getVirtualCores(), is(YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_VCORES));
+		}
```

Comment:
I think we should also verify that after the container is allocated the request is removed.
Or we can make it a separate test case.